lib/bloc/location/location_state.dart

```diff
@@ -2,12 +2,18 @@ part of 'location_bloc.dart';
 
 abstract class LocationState {}
 
+class InternetDisconectedState extends LocationState {}
+
+class RequestPermissionState extends LocationState {}
+
 class LocationInitial extends LocationState {}
 
 class LocationServiceDisableState extends LocationState {}
 
 class LocationPermissionNotGrantedState extends LocationState {}
 
+class LocationPermissionDeniedState extends LocationState {}
```

Comment:
`LocationPermissionDeniedState` and `LocationPermissionNotGrantedState` are the same meaning right? can u reuse `LocationPermissionNotGrantedState`, or change it to be `LocationPermissionDeniedState`